packages/manager/apps/web/client/app/hosting/multisite/hosting-multisite.controller.js

```diff
@@ -128,6 +134,98 @@ angular
                 );
               }
             });
+            return WucUser.getUser();
+          })
+          .then((user) => {
+            $scope.user = user;
+
+            const getNichandleAttached = (
+              diagnosticSuffix,
+              domain,
+              nichandle,
+            ) => {
+              return HostingDomain.getServiceInfosForZone(domain.name).then(
+                (serviceInfos) => {
+                  const { contactTech, contactAdmin } = serviceInfos;
+
+                  const isDnsAttachedToNic =
+                    nichandle === contactAdmin || nichandle === contactTech;
+                  return {
+                    diagnosticState: diagnosticSuffix[domain.name],
+                    isDnsAttachedToNic,
+                  };
+                },
+              );
+            };
+
+            const promises = $scope.domains.list.results.map((domain) => {
+              return $q
+                .all([
+                  HostingDomain.getDigStatus(
+                    $stateParams.productId,
+                    domain.name,
+                  ),
+                  HostingDomain.getZoneLinked(domain.name),
+                ])
+                .then(([digStatus, zoneLinked]) => {
+                  const isDnsExternal = Object.keys(zoneLinked).length === 0;
+                  $scope.optionsForModal[domain.name] = {
+                    ip: $scope.hosting.clusterIp,
+                    domain: domain.name,
+                    dns: $scope.hosting.serviceName,
+                    nic: $scope.user.nichandle,
+                    isDnsExternal,
+                  };
+
+                  if (
+                    isDnsExternal &&
+                    domain.name !== $scope.hosting.serviceName
+                  ) {
+                    $scope.optionsForModal[domain.name] = {
+                      ...$scope.optionsForModal[domain.name],
+                      diagnosticState: DIAGNOSTIC_STATE.ERROR,
+                      isDnsAttachedToNic: false,
+                    };
+                    return;
+                  }
+
+                  if (Object.keys(digStatus.records).length === 0) {
+                    $scope.diagnosticSuffix[domain.name] =
+                      DIAGNOSTIC_STATE.ERROR;
+                    getNichandleAttached(
+                      $scope.diagnosticSuffix,
+                      domain,
+                      $scope.user.nichandle,
+                    ).then((nicProperties) => {
+                      $scope.optionsForModal[domain.name] = {
+                        ...$scope.optionsForModal[domain.name],
+                        zone: zoneLinked.zone,
+                        diagnosticState: nicProperties.diagnosticState,
+                        isDnsAttachedToNic: nicProperties.isDnsAttachedToNic,
+                      };
+                    });
```

Comment:
I think the then part can be moved to `getNichandleAttached` as it is the same for the two calls you make to it

Reply:
Right